woocommerce/woocommerce
# Fix selection of term after clicking the value name

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR addresses an issue with selecting term attributes. Previously, when clicking a term name, it would not be selected as expected.
The proposed changes fix this behavior. The method [isEventOutside](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/components/src/experimental-select-control/select-control.tsx#L243) was erroneously returning true, while the event was actually inside the component.

Closes #39456.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this branch.
2. Install and activate all the required plugins.
3. Add a sample product CSV file or create a couple of global attributes with terms (under Product > Attributes).
4. Go to Woocommerce > Setting > Advanced page.
5. Enable the "New product editor" option.
6. Go to the Products > Add new" page. 
7. Click on the "Add  Attribute" button.
8. Select an Attribute, then click on the dropdown option for value.
9. Try clicking on the text of any attribute value. The term should now be selected as expected.

![Screen Capture on 2023-08-03 at 18-56-39](https://github.com/woocommerce/woocommerce/assets/1314156/821170cd-5180-4cbb-b38f-ed36f16f31d9)




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
